### ISSUE TITLE
Core: Prevent duplicate IDs in DOM

### ISSUE THREAD

**Issue by melloware** [primefaces/primefaces#8765]
### Describe the bug

PrimeFaces has a few places where duplicate IDs are present in the DOM.  This issue is to track all of them in 1 spot.  Most of the time its in places where PF uses `UIColumn` to iterate over some advanced rendering but not always.

**Why is it bad?**
https://softwareengineering.stackexchange.com/questions/127178/two-html-elements-with-same-id-attribute-how-bad-is-it-really

**Components Affected:**
- [x] AutoComplete (https://github.com/primefaces/primefaces/issues/7763
- [x] SelectOneMenu (https://github.com/primefaces/primefaces/issues/7767)
- [x] OrderList (https://github.com/primefaces/primefaces/issues/3781)
- [ ] DataTable RowExpansion (https://github.com/primefaces/primefaces/issues/7765)
- [ ] Diagram (https://github.com/primefaces/primefaces/issues/9551)

- [x] SelectCheckBoxMenu (https://github.com/primefaces/primefaces/issues/4696)

**Possible Solutions:**
1. Just ignore it until browser doesn't support it anymore
2. Commit a fast workaround (https://github.com/primefaces/primefaces/pull/8761)
3. Do it the right JSF way, which takes a long time using `UIRepeat / UINamingContainer`

### Reproducer

Great example from https://github.com/primefaces/primefaces/issues/7767 on how to test with SelectOne and use a script to check for duplicate ID's.

```xml
<h:form id="form">
	<p:selectOneMenu id="menu" var="v">
		<f:selectItems value="#{testView.selectItems}" />
		<p:column>
			<h:outputText id="item_id" value="#{v.id}" />
		</p:column>
		<p:column>
			<h:outputText id="item_name" value="#{v.name}" />
		</p:column>
	</p:selectOneMenu>
</h:form>

<script>
	function checkDuplicateIds() {
		setTimeout(function() {
			var ids = {};
			$('[id]').each(function() {
				var id = this.id;
				if (id &amp;&amp; ids[id]) {
					ids[id].push(this);
					console.error('Duplicate id: ' + id + ' in elements', ids[id]);
				} else {
					ids[id] = [this];
				}
			});
		}, 100);
	}
	checkDuplicateIds();
</script>
```

### Expected behavior

No components should generate duplicate ID's and all ID's in the DOM should be unique.

Client ids should probably include iteration, like with UI "x:1:y", "x:2:y"...

### PrimeFaces edition

Community

### PrimeFaces version

11.0.0

### Theme

ANY

### JSF implementation

All

### JSF version

ALL

### Browser(s)

ALL is

**Comment by Bukama** [comment]
It's even bad cause the `ids` are used for references of labels/error messages and are core element for screen readers. So according to your discussion in the PR about performance / duplicate IDs: "First make it right (no duplicate Ids) then make it fast" would be my point.

**Comment by melloware** [comment]
Totally agree.  We probably need to fix this with the difficult solution of naming container and ui repeat etc.

**Comment by rider87** [comment]
@melloware I cannot follow this request / issue... 
This issue (p:diagram) has been closed  https://github.com/primefaces/primefaces/issues/9551 ...
This issue here (https://github.com/primefaces/primefaces/issues/8765) has now also closed?
Is this issue (p:diagram) now solved - I´m confused.

**Comment by melloware** [comment]
Its fixed now.

**Comment by rider87** [comment]
wow, you´re the best. Thanks a lot. 
When do you think it will be ready in Maven?

**Comment by melloware** [comment]
sometime this week for 13.0.6

**Comment by melloware** [comment]
@rider87 you can also try 14.0.0-SNAPSHOT right now: https://github.com/primefaces/primefaces?tab=readme-ov-file#snapshot
